### PR TITLE
LPD-26845, LPD-25961, LPD-25962, LPD-26854, LPD-26856 Autosave should save all the changes made on the content

### PR DESCRIPTION
**Pull request by NemethNorbert**
Hey,

# Required FeatureFlags:

```
feature.flag.LPD-11228=true
feature.flag.LPS-141392=true
feature.flag.LPD-15596=true
```

# What is this trying to solve?

I sent this in one PR as all of them is connected to resolve the issues under [LPD-24043](https://liferay.atlassian.net/browse/LPD-24043) and [LPD-24020](https://liferay.atlassian.net/browse/LPD-24020)

Explain *what* you're trying to do, without talking about *how* you're doing it. Here, brevity is a virtue, don't get into too much detail. An overview should be more than enough.

# How am I fixing it?

I introduced an errorState for the lock indicator, in case we cannot save the article as draft due to some missing required field like the title, we show an error.

In order to trigger this, I introduced frontend validation to the fields when saving the article as draft, I trigger the same validation what we would do when publishing the web content.

The friendlyURL is a special field, when we first save the web content we create a friendlyURL for the web content. This information is provided now as a parameter in the url, so we can update the field once the first save as draft action happens, so we can avoid friendlyURL exceptions on subsequent update article actions

In order to be able to publish the Article, that was already saved as draft, I had to update the generated articleId within SaveButton component as well, this ensures an update instead of a duplication with a new articleId.

# How can you verify that it works?
"Follow the steps described in (https://liferay.atlassian.net/browse/LPD-24043) and [LPD-24020](https://liferay.atlassian.net/browse/LPD-24020)"

Run the functional tests

### CODE REVIEW

File: modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java [hunk 1]
```diff
@@ -715,9 +715,8 @@ private String _getLocaleDisplayNames(Locale locale, Locale... locales) {
 	}
 
 	private String _getSaveAndContinueRedirect(
-			ActionRequest actionRequest, JournalArticle article,
-			String redirect)
-		throws Exception {
+		String actionName, ActionRequest actionRequest, JournalArticle article,
```
Comment (AliciaGarciaGarcia) [inline modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java:718]: ActionRequest/ActionResponse normally goes first place

**Comment by AliciaGarciaGarcia** [comment]
Hi @NemethNorbert 

maybe this is not intended or not but if you work on the WC without a title there is an error on the WC


https://github.com/liferay-content-management/liferay-portal/assets/47524751/451ab6a1-25b6-4f47-92fd-3f354d691a58

File: modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java (hunk 1 above)
Comment (jkappler) [reply modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java:718]: @AliciaGarciaGarcia, isn't it alphabetically ordered?

Comment (AliciaGarciaGarcia) [reply modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java:718]: @jkappler knows best

Comment (jkappler) [reply modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java:718]: No, lets go with the recommendation of @AliciaGarciaGarcia. @NemethNorbert, can you change the order please (sorry, my bad for specifying it like this).

**Comment by NemethNorbert** [comment]
> Hi @NemethNorbert
> 
> maybe this is not intended or not but if you work on the WC without a title there is an error on the WC
> 
>  Compartir.pantalla.-.2024-05-27.17_54_15.mp4

That's the frontend validation that runs when the autosave would happen.

File: modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java (hunk 1 above)
Comment (NemethNorbert) [reply modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java]: I reordered it

Comment (jkappler) [reply modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/UpdateArticleMVCActionCommand.java]: Thanks @NemethNorbert!

**Review by jkappler — APPROVED** [review]

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_article.jsp [hunk 2]
```diff
@@ -145,6 +145,15 @@ journalEditArticleDisplayContext.setViewAttributes();
 					<div class="c-gap-3 form-group-sm journal-article-button-row mb-0 tbar-section text-right">
 						<c:choose>
 							<c:when test='<%= FeatureFlagManagerUtil.isEnabled("LPS-141392") %>'>
+								<div class="align-items-center d-none mx-3 small text-danger" id="<portlet:namespace />lockErrorIndicator">
+									<liferay-ui:message key="alert-helper-error" />
+
+									<clay:icon
+										cssClass="ml-2"
```
Comment (ambrinchaudhary) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_article.jsp:152]: please add class `mt-0 ` so the icon is properly aligned

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]
Just a minor change request @NemethNorbert. 
Another missing thing: when the error messag appears because the title is empty, once it is filled this message is not being removed (we need to remove it as well as the message Error! is) 

IMHO it is not good from UX perspective to start the autosave if the title is not filled. what @AliciaGarciaGarcia mentioned also looks strange to me. In Blogs, we do that way, autosave is not triggered until title is filled. But this is something that maybe PO must decide,so for know, once that class is added and the errror message is removed, the pull is approved from my side.

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_article.jsp (hunk 2 above)
Comment (NemethNorbert) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_article.jsp]: I force pushed the requested changes

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by ambrinchaudhary** [comment]
ci:forward

**Pull request closed without merging**